commands/remove.rb

```diff
@@ -13,9 +13,18 @@ def self.remove(pkg, verbose)
       return
     end
 
-    # Don't remove any of the packages ruby (and thus crew) needs to run.
-    if CREW_ESSENTIAL_PACKAGES.include?(pkg.name)
-      puts "Refusing to remove essential package #{pkg.name}.".lightred
+    # Determine dependencies of packages in CREW_ESSENTIAL_PACKAGES, as
+    # those are needed for ruby and crew to run, and thus should not be
+    # removed.
+    essential_deps = CREW_ESSENTIAL_PACKAGES.flat_map.uniq do |pkg|
+      Package.load_package("#{pkg}.rb").get_deps_list
+    end
+    if essential_deps.include?(pkg.name)
```

Comment:
We should check for the package that we are currently operating also, not just its dependencies

Reply:
ok hopefully this makes more sense:
```ruby
    # Determine dependencies of packages in CREW_ESSENTIAL_PACKAGES and
    # their dependencies, as those are needed for ruby and crew to run,
    # and thus should not be removed.
    essential_deps = CREW_ESSENTIAL_PACKAGES.flat_map { |essential_pkg| Package.load_package("#{essential_pkg}.rb").get_deps_list }.push(*CREW_ESSENTIAL_PACKAGES).uniq.sort
    essential_deps2 = essential_deps
    loops = 0
    until loops == 5
      essential_deps = essential_deps2.flat_map { |essential_pkg| Package.load_package("#{essential_pkg}.rb").get_deps_list }.push(*essential_deps).uniq.sort

      break if essential_deps == essential_deps2
      essential_deps2 = essential_deps
      loops += 1
    end

    if essential_deps.include?(pkg.name)
      puts <<~ESSENTIAL_PACKAGE_WARNING_EOF.gsub(/^(?=\w)/, '  ').lightred
        #{pkg.name.capitalize} is considered an essential package needed for
        Chromebrew to function and thus cannot be removed.
      ESSENTIAL_PACKAGE_WARNING_EOF

      return
    end
```

Reply:
Let me know if there is a more efficient way of doing this.